sqlmesh/core/snapshot/definition.py

```diff
@@ -794,6 +795,43 @@ def version_get_or_generate(self) -> str:
         """Helper method to get the version or generate it from the fingerprint."""
         return self.version or self.fingerprint.to_version()
 
+    def is_valid_start(
+        self,
+        start: t.Optional[TimeLike],
+        snapshot_start: t.Optional[TimeLike] = None,
+        execution_time: t.Optional[TimeLike] = None,
+    ) -> bool:
+        """Checks if the given start and end are valid for this snapshot.
+        Args:
+            start: The start date/time of the interval (inclusive)
+            snapshot_start: The start date/time of the snapshot (inclusive)
+        """
+        # The snapshot may not have a start defined. If so we use the provided snapshot start.
+        if self.depends_on_past and start:
+            if not snapshot_start:
+                raise SQLMeshError("Snapshot must have a start defined if it depends on past")
+            start_ts = to_timestamp(self.model.cron_floor(start))
```

Comment:
I think this should be `self.model.interval_unit.cron_floor(start)`. `self.model.cron_floor(start)` is based on model's cron which is not always representative of the actual interval unit.

Reply:
Nice catch!